### ISSUE TITLE
Consider using existing shared libs?

### ISSUE THREAD

**Issue by SchrodingerZhu** [koka-lang/koka#205]
The PCRE examples indicate that Koka seems to scan for static libs (especially for those from vcpkg) on default. Is it possible to consider build the lib with shared lib support if it is feasible in the given environment?

**Comment by daanx** [comment]
Yes, at the moment only static linking is supported; There is still work to be done to make even static linking better (and I hope it gets improved once I integrate libuv). For now, you can specify in an `extern import`:

- `library`: the library basename (pcre2-8). If found, it is copied to the output directory and linked.
- `vcpkg`: the vcpckg name: if this is given, koka will use vcpckg to get and build the the package and get the library from there. This is very convenient as the koka library author does not need to distribute the C library with it.  
- `syslib`: for system libraries (like `pthread`) which only links with it.

It would be nice to have Conan & pcconfig support as well for now I am holding back a bit as I guess what Koka really needs is a package manager. The goal is though to be opinionated and have koka builds be seamless where external libraries get build/found automatically. 

I am very hesitant about dynamic linking as that has many issues on many platforms and makes deterministic builds difficult. Given the limited resources, for now, I would like to pursue a static linking approach only for Koka but make it as easy as possible.
A nice benefit is that a Koka binary will only depend on libc and can be run without dynamic link failures at deployment.

**Comment by locriacyber** [comment]
Here because of this error when compiling.
```
warning: unable to find C library: pcre2-8
   hint: provide "--cclibdir" as an option, or use "syslib" in an extern import?
failure during code generation:
  unable to find C library pcre2-8
  library search paths: ["/usr/local/lib","/usr/lib","/lib"]
```

Is it possible to bundle pcre source code with koka?

About static linking, does koka use musl libc? Even most "static" libraries depend on external libc symbols.
